File: Programming Basics with Python/04 PB Python for loop/More exercises/02_Hospital.py
days = int(input())
doctors = 7
treated_patients = 0
untreated_patients = 0
for period in range(1, days + 1):
    patients_per_day = int(input())
    if period % 3 == 0:
        if untreated_patients > treated_patients:
            doctors +=1
    if patients_per_day <= doctors:
        treated_patients += patients_per_day
    else:
        treated_patients += doctors
        untreated_patients += (patients_per_day - doctors)
print(f'Treated patients: {treated_patients}.')
print(f'Untreated patients: {untreated_patients}.')

File: JS Basics/04 For-Loop - More Exercises/02. Hospital.js
function hospital(numbers) {
    let n = Number(numbers[0]);
    let doctors = 7;
    let dayCounter = 0;
    let treated_patients = 0;
    let untreated_patients = 0;
    for (let i = 1; i <= n; i++) {
        let patients = Number(numbers[i]);
        dayCounter += 1;
        if (dayCounter % 3 == 0) {
            dayCounter = 0;
            if (untreated_patients > treated_patients) {
                doctors += 1;
            }
        }
        if (patients > doctors) {
            treated_patients += doctors;
            untreated_patients += patients - doctors;
        }
        else {
            treated_patients += patients;
        }
    }
    console.log(`Treated patients: ${treated_patients}.`)
    console.log(`Untreated patients: ${untreated_patients}.`)
}
hospital([3, 7, 7, 7])